fsspec/implementations/http.py

```diff
@@ -358,9 +358,10 @@ def _open(
         kw = self.kwargs.copy()
         kw["asynchronous"] = self.asynchronous
         kw.update(kwargs)
-        size = size or self.info(path, **kwargs)["size"]
+        info = self.info(path, **kwargs)
```

Comment:
We really don't want to do this unless necessary. I think that, in the case that `size` is provided, we can ssume that ranges should work and we can do random access. This moves the responsibility to the calling code.

Reply:
Will change to:

```suggestion
        partial = True
        if size is None:
            info = self.info(path, **kwargs)
            size = info["size"]
            partial = info.get("partial", partial)
        session = sync(self.loop, self.set_session)
        if block_size and size and partial:
```